docs/discussion/Ratify v2 design scope.md

```diff
@@ -0,0 +1,165 @@
+Proposal: A more lightweight and extensible framework for Ratify v2
+=======================
+Author: Binbin Li
+
+## Table of Contents
+- [Proposal: A more lightweight and extensible framework for Ratify v2](#proposal-a-more-lightweight-and-extensible-framework-for-ratify-v2)
+  - [Table of Contents](#table-of-contents)
+  - [Background](#background)
+  - [Goal](#goal)
+  - [Scope](#scope)
+    - [Refactoring](#refactoring)
+      - [Cache Refactoring](#cache-refactoring)
+      - [Policy Provider Refactoring](#policy-provider-refactoring)
+      - [Configuration Refactoring](#configuration-refactoring)
+      - [Reorganize the Codebase](#reorganize-the-codebase)
+    - [Deprecations](#deprecations)
+      - [CertificateStore](#certificatestore)
+      - [Legacy Cosign Verifier](#legacy-cosign-verifier)
+      - [License Checker Verifier](#license-checker-verifier)
+      - [`Name` and `Type` fields in verifierReport/configuration](#name-and-type-fields-in-verifierreportconfiguration)
+    - [New Features](#new-features)
+    - [Documentation Update](#documentation-update)
+    - [Unit Test Improvement](#unit-test-improvement)
+
+## Background
+Ratify has reached its first major release in over a year and has been widely adopted by users at scale. However, there are known limitations in the current design and implementation of Ratify v1 make it challenging to extend functionality. This document outlines these known limitations and proposes solutions for a more lightweight and extensible framework.
+
+## Goal
+
+The goal of the v2 design is to improve the current design and implementation to better meet the requirements of the customers.
+
+1. Deprecate outdated features so that customers can have a clear understanding of the features that are still supported.
+2. Refactor the code to make it more maintainable and extensible for new features.
+3. Improve the performance of the system.
+4. Improve the usability of the system.
```

Comment:
Have we considered adding new primitives that wrap/simplify the other components. (e.g introduce a new CRD that synthesizes store + verifier + policy). This is similar to the simplification proposed by @yizha1 so that user is only concerned with a single CRD.

Reply:
yeah, that's good call. I agree that would enhance user experience, but don't have clear idea on it yet. Just wrapping everything up may not be a good solution. And I'm also considering deprecating wrapped CRDs by this new CRD. May need more feedback from @yizha1 as we have discussed it before.